tests/test-dirs/type-enclosing/underscore-ids.t

```diff
@@ -0,0 +1,309 @@
+These tests ensure the stability of identifier reconstruction
+in the presence of underscores.
+
+  $ $MERLIN single type-enclosing -position 3:2 -filename under.ml <<EOF
+  > let _foo = 4.2
+  > let f () : int =
+  >   _foo
+  > EOF
+  {
+    "class": "return",
+    "value": [
+      {
+        "start": {
+          "line": 3,
+          "col": 2
+        },
+        "end": {
+          "line": 3,
+          "col": 6
+        },
+        "type": "float",
+        "tail": "no"
+      },
+      {
+        "start": {
+          "line": 3,
+          "col": 2
+        },
+        "end": {
+          "line": 3,
+          "col": 6
+        },
+        "type": "float",
+        "tail": "no"
+      },
+      {
+        "start": {
+          "line": 3,
+          "col": 2
+        },
+        "end": {
+          "line": 3,
+          "col": 6
+        },
+        "type": "int",
+        "tail": "no"
+      },
+      {
+        "start": {
+          "line": 2,
+          "col": 6
+        },
+        "end": {
+          "line": 3,
+          "col": 6
+        },
+        "type": "unit -> int",
+        "tail": "no"
+      }
+    ],
+    "notifications": []
+  }
+
```

Comment:
I'm a bit confused by all the tests below: I'm sure they're exercising the identifier reconstruction code, but we can't really observe the result, can we?

Reply:
Because all that count is the return type annotation ?
I should define the value first and check the type like the first test I guess.